site/app/views/NavigationView.php

```diff
@@ -303,7 +303,7 @@ private function hasGradeButton(Gradeable $gradeable): bool {
         $im_a_grader = $this->core->getUser()->accessGrading() && $this->core->getUser()->getGroup() <= $gradeable->getMinGradingGroup();
 
         // students can only view the submissions & grading interface if its a peer grading assignment
-        $im_a_peer_grader = $this->core->getUser()->getGroup() === User::GROUP_STUDENT && $gradeable->isPeerGrading();
+        $im_a_peer_grader = $this->core->getUser()->getGroup() === User::GROUP_STUDENT && $gradeable->isPeerGrading() && $this->core->getQueries()->getPeerGradingAssignmentsForGrader($this->core->getUser()->getId());
```

Comment:
I believe we need similar logic when displaying the TA grading page. Per current master, the check just assumes limited access graders == peers.